### PR TITLE
Carousel improvements

### PR DESCRIPTION
**Pull request by gtrufitt**
## What does this change?

- Dots are clickable *but not tabble* as it makes more navigation sense to navigate the stories directly from both a keyboard-nav and screenreader perspective *with this current implementation*
- Can navigate carousel on desktop with trackpad
- Fixes dot numbers so that the last dot is always the last dot
- Adds hover and focus styles to the arrows and removed default outline styling

https://5dfcbf3012392c0020e7140b-yzforzryrh.chromatic.com/?path=/story/components-carousel--headlines

### CODE REVIEW

**Review by liywjl — APPROVED** [review]
👍 looks great

File: src/web/components/Onwards/Carousel/Carousel.tsx [hunk 1]
```diff
@@ -389,10 +428,18 @@ export const Carousel: React.FC<OnwardsType> = ({
                     <Title title={heading} />
 
                     <div className={navIconStyle} data-link-name="nav-arrow">
-                        <button onClick={prev} className={buttonStyle}>
+                        <button
+                            onClick={prev}
+                            aria-label="Move carousel backwards"
```
Comment (oliverlloyd) [inline src/web/components/Onwards/Carousel/Carousel.tsx:433]: 👏

File: src/web/components/Onwards/Carousel/Carousel.tsx [hunk 2]
```diff
@@ -161,28 +165,42 @@ const dotsStyle = css`
...
     ${from.phablet} {
-        display: ${index >= 7 ? 'none' : 'auto'};
+        display: ${index >= totalStories - 1 ? 'none' : 'auto'};
     }
 
     ${from.desktop} {
-        display: ${index >= 6 ? 'none' : 'auto'};
+        display: ${index >= totalStories - 2 ? 'none' : 'auto'};
     }
 `;
```
Comment (oliverlloyd) [inline src/web/components/Onwards/Carousel/Carousel.tsx:194]: If you brought this out into it's own style function then you wouldn't need to pass in `totalStories` to `dotActiveStyle`?

File: src/web/components/Onwards/Carousel/Carousel.tsx [hunk 3]
```diff
@@ -161,28 +165,42 @@ const dotsStyle = css`
...
 
-const dotActiveStyle = (index: number) => css`
-    ${dotStyle(index)};
+const dotActiveStyle = (index: number, totalStories: number) => css`
+    ${dotStyle(index, totalStories)};
```
Comment (oliverlloyd) [inline src/web/components/Onwards/Carousel/Carousel.tsx:197]: As above, if you extracted the display styling out you could remove the need for `totalStories` here. Possibly better?

File: src/web/components/Onwards/Carousel/Carousel.tsx [hunk 4]
```diff
@@ -161,28 +165,42 @@ const dotsStyle = css`
...
     margin-right: ${space[1]}px;
 
+    &:hover,
+    &:focus {
+        background-color: ${palette.neutral[86]};
+        outline: none;
+    }
+
     /* This is a bit of a hack for the test, while we think of better UX here.
     It's very fragile to things like carousel item count.*/
```
Comment (oliverlloyd) [inline src/web/components/Onwards/Carousel/Carousel.tsx:186]: Is this comment still true? Will this css still work well if `totalStories` is 1 or 20?

**Review by oliverlloyd — APPROVED** [review]
+1 for thinking about keyboard users

**Comment by gtrufitt** [comment]
This is gtg on Monday

**Pull request merged**